packages/x-date-pickers-pro/src/SingleInputDateRangeField/useSingleInputDateRangeField.ts

```diff
@@ -70,6 +70,6 @@ export const useSingleInputDateRangeField = <TDate, TChildProps extends {}>({
     valueManager: rangeValueManager,
     fieldValueManager: rangeFieldValueManager,
     validator: validateDateRange,
-    supportedDateSections: ['year', 'month', 'day'],
```

Comment:
I think this new abstraction is simpler to understand and easier to reason with if we have logic to apply like the one of this PR.